### DISCUSSION TITLE
First day with OSMand and what a disaster

### DISCUSSION THREAD

**Discussion by GusGF** [osmandapp/OsmAnd#15123] [Q&A]
Okay not entirely as it's a walking track I've done many times before luckily but here are the issues I had

- Voice would give me directions for a series of turnings I'd yet to reach as if the voice prompts and map were out of sync. Towards the end of the walk, I realised this as the app gave me instructions to follow immediately even though I was 3-400 metres from where it was related to.
- Telling me to turn right or left on a straight track with no turns (probably related to above)
- Bends in the route being interpreted as turns

I imagine a lot of the issues are tied to the problem I mentioned already but it's baffling that this should happen at all!

**Comment by tommycrock** [comment]
Hi GusGF.
Can you tell us where the route for the instructions came from (e.g. a gpx file from a website, pinning a destination and selecting navigation, etc.), whether you have simulate navigation turned on (menu, navigation, settings, simulate navigation (bottom option)), your phone, Android OS version and osmand version? These will probably help people understand what's going on. Thanks.

**Reply by GusGF** [reply]
Apologies for not replying sooner. It was a walking route I'd drawn over walking tracks in the app on my mobile. I don't even know how to turn on simulated navigation or have not purposely used it before. The walk lasted 4 hours with about 3 hrs of actual walking.

Samsung A52s
Android v12
OSMand+ 4.2.7

**Comment by barteis** [comment]
It sounds like you're following the GPX, not the underlying map.  This may help. https://vimeo.com/654264099

**Reply by GusGF** [reply]
Just watched it and it wasn't that clear how he was navigating from one screen to another or how at the end how he'd edited the route to his requirements, wish he'd slowed down and explained what he was doing. For the walk he seemed to use 'attach to road' and for the bicycle 'route between points'. But no explanation or insight as to why?!

Incidentally when I opened my track and used the option 'attached to road' things got far worse. I've now got navigation which for the most part seems to be as the crow flies i.e. straight lines between invisible points and not sticking to the walking track/path at all!

**Comment by barteis** [comment]
The first case shows how to adapt a GPX when you are at the point of departure.  The second case shows how to preview a GPX using the Plan route feature ahead of time.  in both cases, you are routing between points, that is, creating a route  based on the GPX that takes into account the underlying map.  I'm not sure why you are getting straight lines instead.  If you can provide a URL to the GPX you are working with (I don't think Github will take it directly), along with the name of the map you're using, I'm happy to take a look.  

You might also try the third method, using the Plan route feature to trace over your GPX using the navigation profile of your choice.

**Reply by GusGF** [reply]
Hi Bart,
Am I right in assuming a gpx file is simply a breadcrumb trail or series of points and does or does not include directions?
Hence do all GPX files have to be connected to the underlying map being used by the mapping/nav app being used?

Okay with the motorcycle gpx you've used the option 'attach to the roads' (using the parameters set by your current profile)
What's the difference between a route and a track as I've noticed in OSMand you plan a route but follow a track?

With the planned bicycle track you attached it to the underlying map in the app but this time using the option 'route between points'. Which according to OSMand does exactly the same thing as above. I'm confused why use different terminology? '**attach to the roads**' vs '**route between points**', what is the difference?

So when you save this cycle track as a new track what are you achieving? Are you embedding the turn-by-turn directions? Is this also a GPX file you can share, but with directions? So if a GPX can embed directions how do you know if it does or doesn't? Why wasn't this automatically done when creating the route or should I say track?

Sorry for all the questions and you can just tell me to go and read your site https://sites.google.com/view/bartetutorials/home if your confident I'll get the answers. Thanks for trying to help people like me :)

**Comment by barteis** [comment]
You're right: the terms are confusing.  Osmond's use is not consistent with the official GPX definitions.  I'm inconsistent as well. 

Per Wikipedia's GPX article: "Conceptually, tracks are a record of where a person has been and routes are suggestions about where they might go in the future. For example, each point in a track may have a timestamp (because someone recorded where and when they were there), but the points in a route are unlikely to have timestamps (other than estimated trip duration) because route is a suggestion which might never have been traveled."

My video is on downloading GPX _routes_ from the web for use with OsmAnd.  There's no assumption these GPXs contain directions.  If you use OsmAnd to follow one, the app might only know whether the route is straight, veering left or veering right. On the other hand, if you create a GPX route with OsmAnd, it will use the underlying map to embed street and even hiking path names. 

An OsmAnd-created GPX route used on OsmAnd will provide true GPS turn-by-turn directions.  A GPX imported into OsmAnd may or may not.  

The video shows two ways of producing OsmAnd-friendly GPXs: from the beginning of your walk or ride, or, using the Plan route feature ahead of time.  Which ever way you choose, you will eventually get to a screen that says "Route between the points".  That's where OsmAnd produces its own GPX using the GPX you provided as a basis.  (The terms "attach to roads" and "route between points" are Osmond's descriptions, not mine.)

Note that you don't need true turn-by-turn directions in every case.  I sometimes hike off-trail.  If I want to return using my exact track (the record of where I've been) for stream crossings, for example, I don't worry about trail names or intersections, because there aren't any.

Here's what I think you should do.  Display your GPX on the screen.  Then use the Plan route feature to trace over that GPX using the walking profile.  It's set with the circular icon on the left.  Doing that, you will see more clearly what OsmAnd is doing.  When you're done, you can "Save the track", which actually saves not a track, but a route.  Whatever you call it, this new GPX should give you more of what you're looking for.  This video might help: https://vimeo.com/637904555

Bart

**Reply by GusGF** [reply]
I planned a walk along a path [GPX file](https://drive.google.com/file/d/1bxcbfL8dimL3hLvOnB5h5_aYDTIxv0Zt/view?usp=sharing) and it got the turn instructions very wrong. Giving the directions out of sync with the map. I just ran the simulation and it doesn't include some of the cock-ups.

In your video for the cycle trip **after** creating the route externally using another app and importing it, you then went into the menus, plan route, open existing track, then route between points. I didn't do this second bit after planning the route in OSMand, I just saved it. Was this wrong?

So what happens if you download a GPX for a long route? You are not expecting someone to sit there tracing over the gpx and then use the above menus to generate proper directions? I'm assuming you would import it, and chose "Route between the points" to link it to the underlying map. Right?

**Comment by barteis** [comment]
If I understand your first question, yes, you have to route between points. 

Long routes: yes, you would do it as you describe.  I suggested the exercise of tracing a route to get a better idea of how OsmAnd works.  You provide the points, OsmAnd routes between them.  You can watch it in real time.   

I used Plan route on the route you linked to produce [a second GPX](https://drive.google.com/file/d/1QACdAUjirwL1Szcp9iTTOYV9kwvHDWFv/view?usp=sharing).  How does it look?

Bart

**Reply by GusGF** [reply]
> If I understand your first question, yes, you have to route between points.

Okay, sorry just to clarify. Even if you planned the route in OSMand on the mobile device??

> Long routes: yes, you would do it as you describe. I suggested the exercise of tracing a route to get a better idea of how OsmAnd works. You provide the points, OsmAnd routes between them. You can watch it in real time.

Okay I see, thanks.

> I used Plan route on the route you linked to produce [a second GPX](https://drive.google.com/file/d/1QACdAUjirwL1Szcp9iTTOYV9kwvHDWFv/view?usp=sharing). How does it look?

When you say plan route, do you mean you got OSMand to route between points? I'll only really know when I'm next out as simulation mode doesn't completely replicate what instructions I'll get back.

**Reply by barteis** [reply]
If you planned the route in OsmAnd, it should work as is on OsmAnd.  I'm only referring here to GPXs imported into OsmAnd that were created elsewhere.  

And right, I took your route, and used the Plan route feature to create a second GPX.  Plan route/Open existing track (or Import track) / Option/Route between points/<pick a profile>/Save as new track.  I look forward to hearing how it does in the field!

Bart

**Reply by GusGF** [reply]
Yes I know, that is odd as it should have worked off the bat. I'm trying out another app called Locus as well but will get back to you. Thanks for trying to help I'm sure you've got better things to be doing.
